spec/ParseLiveQuery.spec.js

```diff
@@ -115,6 +115,95 @@ describe('ParseLiveQuery', function () {
     });
   });
 
+  it('can handle mutate beforeSubscribe query', async done => {
+    await reconfigureServer({
+      liveQuery: {
+        classNames: ['TestObject'],
+      },
+      startLiveQueryServer: true,
+      verbose: false,
+      silent: true,
+    });
+    Parse.Cloud.beforeSubscribe(TestObject, request => {
+      const query = request.query;
+      query.equalTo('yolo', 'abc');
+    });
+
+    const object = new TestObject();
+    await object.save();
+
+    const query = new Parse.Query(TestObject);
+    query.equalTo('objectId', object.id);
+    const subscription = await query.subscribe();
+
+    subscription.on('update', () => {
+      fail();
+    });
+    object.set({ foo: 'bar' });
+    await object.save();
+    setTimeout(async () => {
+      done();
+    }, 1000);
+  });
+
+  it('can return a new beforeSubscribe query', async done => {
+    await reconfigureServer({
+      liveQuery: {
+        classNames: ['TestObject'],
+      },
+      startLiveQueryServer: true,
+      verbose: false,
+      silent: true,
+    });
+    Parse.Cloud.beforeSubscribe(TestObject, () => {
+      const query = new Parse.Query(TestObject);
+      query.equalTo('foo', 'yolo');
+      return query;
```

Comment:
Do I understand it correctly, that it is possible to return the query but not necessary? That seems strange.

Reply:
I agree

Reply:
@dplewis Didn't I see just a second ago a review comment from you regarding `toJSON`? Did that disappear because I started a review or am I mixing something up here.

Reply:
I deleted it as it wasn't helpful.

Reply:
I added it so you can return a different query if required. Is this unnecessary?

Reply:
To return a different query I would set `request.query`. Giving the additional option of returning the query unnecessarily occupies the return value of the `beforeSubscribe`, which we could to use for something else in the future.

Reply:
Thank you for the feedback @mtrezza, I have removed the return value from `beforeSubscribe` and changed the test case to set `request.query`.